shell/src/main/java/alluxio/cli/Version.java

```diff
@@ -29,6 +47,34 @@ private Version() {} // prevent instantiation
    * @param args the arguments
    */
   public static void main(String[] args) {
-    System.out.println(RuntimeConstants.VERSION);
+    CommandLineParser parser = new DefaultParser();
+    Options opts = new Options().addOption(REVISION_OPT);
+    try {
+      CommandLine cl = parser.parse(opts, args);
+      String revision = "";
+      if (cl.hasOption(REVISION_OPT.getOpt())) {
+        String r = cl.getOptionValue(REVISION_OPT.getOpt());
+        int revisionLen = r == null ? ProjectConstants.REVISION.length() :
+            Integer.parseInt(r);
+        revision = String.format("-%s", ProjectConstants.REVISION.substring(0,
+            Math.min(revisionLen, ProjectConstants.REVISION.length())));
+      }
+      System.out.println(String.format("%s%s", RuntimeConstants.VERSION, revision));
```

Comment:
nit: `System.out.println("%s%s", RuntimeConstants.VERSION, revision);` ?

Reply:
That method signature doesn't exist?

Reply:
Did you mean `System.out.printf`?

Reply:
you are right. nvm